### PR TITLE
Improved UX by displaying hamburger instead of back arrow in mobile header on all pages

### LINKED ISSUE #226: Add hamburger button to menu on mobile topic pages

### LINKED ISSUE #1992: Add hamburger button to menu on mobile topic pages
## Feature Request

**Is your feature request related to a problem? Please describe.**
When on mobile, the hamburger menu slider button appears on tag pages, but not on topic pages. This is annoying, because the menu should be accessible on all pages.

**Describe the solution you'd like**
Add the hamburger menu button to the right of the back-arrow on mobile for topic pages.

**Justify why this feature belongs in Flarum's core, rather than in a third-party extension**
The solution is trivial, and increases ease-of-use for all mobile flarum users.

### PR DESCRIPTION
**Pull request by askvortsov1**
**Fixes flarum/issue-archive#226 **

**Changes proposed in this pull request:**
As of right now, the mobile header displays a back arrow at the top right corner on some pages (specifically discussion pages). This adds no actual value to the user:

- While on non-mobile devices hovering here would bring out the discussion list pane, this is not supported on mobile (and rightly so as it would add unnecessary complexity).
- The only other hypothetical benefit of the back button on mobile is that the button brings the user to the homepage, as opposed to the browser's back button, which might close a window (if the page was opened via a link). However, the hamburger menu more than replaces this functionality: by convention, users know that clicking on the site logo/title in the slideout nav menu will bring them to the homepage (which it does).

In summary, this change makes it much more convenient for users to access their account info/other navbar items, removes a somewhat redundant element, and does not diminish aesthetic.

**Reviewers should focus on:**
If there's no logo, should we add a fontawesome home icon next to the forum title in the mobile navbar?

**Screenshot**
https://i.imgur.com/4MXvXQR.png
https://i.imgur.com/tdUw1vZ.png

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.

### CODE REVIEW

**Comment by franzliedke** [comment]
Hey, we're really close to our feature freeze for beta.12, so I will put this on our internal project board for beta.13. I want the team to have ample time to think about this.

Thanks for the PR! :heart:

**Comment by askvortsov1** [comment]
Absolutely, no rush! Thanks! Should I close the related issue for now, or keep both?

**Comment by franzliedke** [comment]
Hmm, I think I'll do the same thing I did for your other PR and close the PR in favor of discussion in the ticket. Consider it temporary, until a decision has been made.
